app/api/webcal_api.rb

```diff
@@ -81,7 +81,8 @@ class WebcalApi < Grape::API
             projects: { user_id: webcal.user_id },
             units: { active: true }
           )
-          .where('? BETWEEN units.start_date AND units.end_date', Time.zone.now)
+          .where('tasks.project_id is null or tasks.project_id = projects.id')   # eager_load(:tasks) only tasks of :projects
```

Comment:
This is effectively,

```sql
SELECT *
FROM tasks_definitions td
INNER JOIN projects p ON
    ...
LEFT JOIN tasks t ON
    t.task_definition_id = td.id
WHERE
    (t.project_id is null or t.project_id = projects.id)
```

Is there a Rails equivalent of the following?

```sql
SELECT *
FROM tasks_definitions td
INNER JOIN projects p ON
    ...
LEFT JOIN tasks t ON
    t.task_definition_id = td.id
    AND t.project_id = p.id
```

Reply:
I've done this elsewhere using `joins('LEFT OUTER JOIN tutorials ON tutorial_enrolments.tutorial_id = tutorials.id AND (tutorials.tutorial_stream_id = task_definitions.tutorial_stream_id OR tutorials.tutorial_stream_id IS NULL)').`

Reply:
Not sure this would be beneficial. In the end you are just creating a query here to report data.